### ISSUE TITLE
ui continuously refreshed 

### ISSUE THREAD

**Issue by mauriziocescon** [ewhauser/angular-tracing#43]
Hi! 

I've integrated the lib in one of the projects I'm working on. Everything works fine, but I've noticed one huge side-effect: the ui is refreshed continuously. 

Here how to reproduce on heroes-villains what I'm experiencing: 

```
// angular-tracing/apps/heroes-villains/src/app/app.component.html

<div>
  <app-header-bar></app-header-bar>
  <div class="section columns">
    <app-nav class="column is-2"></app-nav>
    <main class="column">
      {{ test }}
      <router-outlet></router-outlet>
    </main>
  </div>
</div>
```

```
// angular-tracing/apps/heroes-villains/src/app/app.component.ts

import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {

  get test(): string {
    console.log(`Date.now: ${Date.now().toString()}`);
    return 'test';
  }
}
```

The `test` getter is called continuously cause some sort of ng-change-detection is continuously triggered. 

Now, this is a huge problem for my application cause we're not implementing OnPush Change detection strategy everywhere: the components-tree is "mark as dirty" too often. 

I wonder: anybody aware of this issue? Any workaround? 

![code](https://user-images.githubusercontent.com/3672842/67283009-2fe00c80-f4d3-11e9-9139-d69e0314d58d.png)

![console](https://user-images.githubusercontent.com/3672842/67283017-32dafd00-f4d3-11e9-81ff-7aa1bd44ab32.png)

**Comment by ewhauser** [comment]
Does this occur in the example application even when the Zipkin module is turned off?

**Comment by mauriziocescon** [comment]
Nope: when Zipkin is commented out, everything works as expected. 

To reproduce it: In addition of the changes above, just add 

```
// angular-tracing/apps/heroes-villains/src/app/app.module.ts 

@NgModule({
  declarations: [AppComponent, AboutComponent],
  imports: [BrowserModule, HttpClientModule, CoreModule, AppRoutingModule, AppStoreModule, /*ZipkinModule.forRoot()*/],
  providers: [/*{ provide: TRACE_MODULE_CONFIGURATION, useFactory: getZipkinConfig }*/],
  bootstrap: [AppComponent]
})
export class AppModule {}
```

```
// angular-tracing/apps/heroes-villains/src/app/heroes/heroes.component.ts

export class HeroesComponent implements OnInit {
  selected: Hero | null;
  heroes$: Observable<Hero[]>;
  message = '?';
  heroToDelete: Hero | null;
  showModal = false;
  // private tracer: LocalTracer;

  constructor(private heroService: HeroService/*, traceRoot: ZipkinTraceRoot*/) {
    this.heroes$ = heroService.entities$;
    // this.tracer = traceRoot.localTracer();
  }
...
```

```
// angular-tracing/apps/heroes-villains/src/app/villains/villains.module.ts

@NgModule({
  imports: [CommonModule, RouterModule.forChild(routes), SharedModule, /*ZipkinModule*/],
  exports: [RouterModule, VillainsComponent],
  declarations: [VillainsComponent, VillainListComponent, VillainDetailComponent]
})
export class VillainsModule {}
```

Of course 

```
  get test(): string {
    console.log(`Date.now: ${Date.now().toString()}`);
    return 'test';
  }```
is called several times at loading time (several ui refreshes). But after that, no more calls.

**Comment by mauriziocescon** [comment]
For your convenience, I've forked the repo and pushed the changes above in a branch. 

Link: [angular-tracing-forked-43](https://github.com/mauriziocescon/angular-tracing/tree/fix/ui-continuously-refreshed-43)

**Comment by mauriziocescon** [comment]
I think the source of problems is the logger definition inside `zipkin.module.ts`: in practice that method is triggering a setInterval inside `zipkin-transport-http` which actually creates a periodic zone-task (zone.js is the mechanism used by angular to update the views). 

Below you find some screenshots. 

So I wonder: is there any particular reason why that code is executed "inside angular"? 

Please take a look at [this](https://stackoverflow.com/questions/47261096/angular-setinterval-cause-component-to-non-stop-update-its-ui) as reference to avoid executing code inside angular. 

![periodic-task-1](https://user-images.githubusercontent.com/3672842/67389102-34232d00-f59a-11e9-9193-07e7b5847d1b.png)
![periodic-task-2](https://user-images.githubusercontent.com/3672842/67389106-35ecf080-f59a-11e9-82df-768a9ccd04df.png)
![periodic-task-3](https://user-images.githubusercontent.com/3672842/67389110-37b6b400-f59a-11e9-824f-46f137cc17be.png)

**Comment by ewhauser** [comment]
Thanks for the reproduction and the fix. Would you mind submitted a PR from your branch? I'll give it a test; but it appears to be the correct fix.

**Comment by ewhauser** [comment]
Thanks - I tested your fix and release `0.0.15`.
